src/observers/observers/models/openai.py

```diff
@@ -64,7 +67,7 @@ def duckdb_schema(self):
             id VARCHAR PRIMARY KEY,
             model VARCHAR,
             timestamp TIMESTAMP,
-            messages STRUCT(role VARCHAR, content VARCHAR)[],
```

Comment:
Yeah very good point. Ollama has this format for images for their [client library for vision](https://ollama.com/blog/llama3.2-vision)

```
"messages": [
    {
      "role": "user",
      "content": "what is in this image?",
      "images": ["<base64-encoded image data>"]
    }
```

Reply:
Or we can rewrite the message structure to follow the following structure? I think they can be used interchangeably according to the typing. WDYT?
{"type": "text", "text": "hello", "image_url": None}
{"type": "text", "text": None, "image_url": "datauri"}

Reply:
These are the message formats. e.g. 
```python
ChatCompletionSystemMessageParam(content=""),
ChatCompletionUserMessageParam(
    content=ChatCompletionContentPartTextParam(text="test")
),
ChatCompletionAssistantMessageParam(content=""),
ChatCompletionUserMessageParam(
    content=ChatCompletionContentPartImageParam(image_url="image")
),
ChatCompletionAssistantMessageParam(content=""),
ChatCompletionUserMessageParam(
    content=ChatCompletionContentPartInputAudioParam(
        input_audio=InputAudio(data="audio", format="wav")
    )
),
ChatCompletionFunctionMessageParam(content=""),
ChatCompletionToolMessageParam(content="")

# the following are equal, more or less

ChatCompletionUserMessageParam(
    content=ChatCompletionContentPartTextParam(text="test")
) 
ChatCompletionUserMessageParam(
    content="test"
)
```

Reply:
Yeah but wouldn't you have image_url etc on every message as None?

Reply:
Yes, that would be the down side, but querying etc would be easier I think.

Reply:
We will have the same problem with SQLite store. It may be better to just store as JSON. It adds a step (CASTING to STRUCT) for querying locally but I think it's a worthy trade off in favor of not introducing a lot of complexity or maintenence